pkg/restapi/webhook/webhook_test.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package webhook
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWebhookDispatcherOneWebhook(t *testing.T) {
+	clientHost := randomURL()
+	clientHandlerPattern := "/webhookListen1"
+
+	subscriberReceivedNotification := make(chan struct{})
+
+	go func() {
+		err := listenAndStopAfterReceivingNotification(clientHost, clientHandlerPattern)
+		if err != nil {
+			require.FailNow(t, err.Error())
+		} else {
+			close(subscriberReceivedNotification)
+		}
+	}()
+
+	if err := listenFor(clientHost); err != nil {
+		t.Fatal(err)
+	}
+
+	StartWebhookDispatcher([]string{fmt.Sprintf("http://%s%s", clientHost, clientHandlerPattern)})
+
+	select {
+	case <-subscriberReceivedNotification:
+		// Pass
+	case <-time.After(5 * time.Second):
+		require.FailNow(t, "samplewebhookclient did not receive a notification")
+	}
+}
+
+func TestWebhookDispatcherMultipleWebhooks(t *testing.T) {
```

Comment:
Use loop instead of repeating each logic 2 times.

Reply:
Done